### DISCUSSION TITLE
Upgrading from a very old version of e107

### DISCUSSION THREAD

**Discussion by WaylonX** [e107inc/e107#5069] [Q&A Updating]
I have a very old e107 system that ran for well over a decade, but now the server is gone and what I have to work with are database dumps of the e107 system and the system files. 

I transferred everything over to a newer server, and not surprisingly, e107 wouldn't run. I'm guessing differences in the mySQL and PHP versions are the reason why, but I'm not really sure.

The e107 system went from a server with PHP 5.x and mySQL 5.? to a server with PHP version 7.3 and MariaDB 10.3.
I'm not 100% certain what version of e107 I have, but maybe this copyright block from one of the pages can shed some light:

|     Copyright (C) 2001-2002 Steve Dunstan (jalist@e107.org)
|     Copyright (C) 2008-2010 e107 Inc (e107.org)
|
|     $URL: https://e107.svn.sourceforge.net/svnroot/e107/trunk/e107_0.7/top.php $
|     $Revision: 12223 $
|     $Id: top.php 12223 2011-05-21 08:01:08Z e107steved $

**In brief, e107 was moved from a legacy system to a modern system, and it won't run.** Given the age of the software I doubt that it'll ever run, so I think maybe an upgrade is the way to resolve this. 

The current state is that e107 is installed, the database and all the files are present. 

I like e107 and would love to run the current version. Is there a way to upgrade this ancient version?

**Comment by Deltik** [comment]
You can find the e107 version in the file `e107_admin/ver.php`.

Upgrading from e107 v0.7 directly to the latest v2 version is not supported, although I'm not sure if it's because it's been proven not to work or if it just hasn't been tested at all.  The suggested upgrade path is v0.7.x → v1.0.4 → v2.x.x.  The upgrade packages are on [the e107 download page](https://e107.org/download).

You may be able to run e107 in a virtual machine with old versions of PHP and MySQL.  Once you can comfortably get on PHP 5.6 and e107 v1.0.4, you should be able to perform an upgrade to e107 v2, which still supports PHP 5.6.  The MySQL version doesn't matter as much.

Third-party plugins and themes have a high chance of being broken on PHP 7 and PHP 8.  With PHP's [`error_log`](https://www.php.net/manual/en/errorfunc.configuration.php#ini.error-log), you may be able to go through the errors one by one to fix them.  Or you can disable such plugins and themes by moving their folders away.

**Reply by WaylonX** [reply]
Thank you, Deltik. Here's what my ver.php file contains, it looks like I have ver 1.0.1, is that correct? 

```
/*
* Copyright (c) 2012 e107 Inc e107.org, Licensed under GNU GPL (http://www.gnu.org/licenses/gpl.txt)
* $URL: https://e107.svn.sourceforge.net/svnroot/e107/trunk/e107_0.7/e107_admin/ver.php $
* $Id: ver.php 12615 2012-03-30 21:00:24Z e107coders $
*
* Version file
*/

if (!defined('e107_INIT')) { exit; }

$e107info['e107_version'] = "1.0.1";
```

1) I don't think I can run it on my remote server in a VM (I've no idea how I'd do that or if it's possible). 

2) I'm a reasonably handy PHP coder, so if there's no upgrade path where I can run it on my server in a VM, what if I installed the latest version of e107 and wrote a script to transfer over the minimum number of fields needed to get the basic pages visible? 

There's only one user account which I think would make that easier if it came to it. I could drop a lot of the plugins without losing too much, but I'd like to get the pages and categories imported or transferred. 

Otherwise, I'm not sure I see a way to bring this back to life. (??)

**Reply by Deltik** [reply]
As long as you have your original site backed up, it won't hurt to try different strategies like extracting the new e107 v2 files over the existing site, installing a fresh copy but importing the database, or in the least convenient case, script something to migrate the old data over.

**Reply by Jimmi08** [reply]
The winning strategy for me was - a fresh install of new e107 (some version from start - f.e 2.1), then switching the database (to old copy), run update process to fix the database, then updating e107 itself.  But true is that I haven't old site for updates to test newer e107 versions for a very long time now, so maybe it works too.  
At first, you should look at the plugin table and check what plugins are installed. Some of them can have issues or will not work. 
The problem with extracting new files over the existing site was outdated plugins.  They were installed and then caused problems. On the other side - sometimes you need to copy folders from the old site (images etc). 

Troubles with updating 0.7 to 1.0.4 - the only issue I am aware of is the admin theme.  If this is the case, you need to switch to the core theme.  But this happens only with some plugins. Check sidebars (left or right menus), if no issues, your theme doesn't have issues.

**Comment by WaylonX** [comment]
I decided to go a different route- e107 is a very capable CMS, and frankly way more than what I need for my application. After considering all the reasonable options, I figured it would be better to start fresh. I had just rebuilt the message board for the site that has e107 so I decided to use an option native to the message board.

I installed a simple CMS that's made to integrate with my message board (Simple Machines Forum) and then spent most of the morning writing a script that would pull out the key bits of info from the e107 DB and put them in the (empty) message board CMS. The hardest part was matching up the fields from the two tables but I only needed about 7 fields from the e107 pcontent table. 

I wrote two scripts (one for the main pcontent table and one for a FAQ add-in) and after a bit of tweaking, I ran them and transferred over about 1,000 pages from e107 altogether (about 300 FAQs and ~650 article pages). 

It's all working fine (better than I expected, lol). The simple portal I used (TinyPortal) is actually fairly capable, nowhere near what e107 can do but it'll work fine for what I need.

Anyway, thank you all for your suggestions; I ran e107 for 10 years but all good things come to an end. 

Cheers
